x-pack/plugins/core_telemetry/server/lib/tasks/visualizations/task_runner.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _, { countBy, groupBy } from 'lodash';
+import moment from 'moment';
+import {
+  IESQueryResponse,
+  IHapiServer,
+  ISavedObjectDoc,
+  ITaskInstance,
+  IVisState,
+  IVisualization,
+} from '../../../../';
+
+interface IVisSummary {
+  type: string;
+  space: string;
+}
+
+/*
+ * Parse the response data into telemetry payload
+ */
+async function getStats(callCluster: (method: string, params: any) => Promise<any>, index: string) {
+  const searchParams = {
+    size: 10000, // elasticsearch index.max_result_window default value
+    index,
+    ignoreUnavailable: true,
+    filterPath: ['hits.hits._id', 'hits.hits._source.visualization'],
+    body: {
+      query: {
+        bool: { filter: { term: { type: 'visualization' } } },
+      },
+    },
+  };
+  const esResponse: IESQueryResponse = await callCluster('search', searchParams);
+  const size = _.get(esResponse, 'hits.hits.length');
+  if (size < 1) {
+    return;
+  }
+
+  // `map` to get the raw types
+  const visSummaries: IVisSummary[] = esResponse.hits.hits.map((hit: ISavedObjectDoc) => {
+    const spacePhrases: string[] = hit._id.split(':');
```

Comment:
Is there already a function for parsing / handling space-scoped IDs? If not, we'll end up with inconsistent mechanisms for parsing IDs and if we need to add more ID modifications in the future, it'd be nice to have fewer places to update.

Reply:
> does there exist some kind of getSpaceNameFromSavedObjectId function shared by the security plugin?

We don't have anything like that currently. At this level, your only options are working with implementation details of Spaces. I'd suggest instead of parsing the document `_id` to use the `namespace` attribute that we apply to saved objects. This will at least give you a 1:1 mapping to space id, with similar rules for the default space:
1) If `namespace` is defined, then it will be set to that saved object's Save ID.
2) If `namespace` is not defined, then that saved object belongs to the `default` space.

As a curious aside, I noticed that this file lives within an `oss_telemetry` folder. Spaces are not an OSS concept, yet they are being reported here.

Reply:
> As a curious aside, I noticed that this file lives within an oss_telemetry folder. Spaces are not an OSS concept, yet they are being reported here.

Good point. We aren't at risk of putting Commercial code into OSS, because all this functionality is in X-Pack. But naming of the plugin was hard.